internal/controller/instaslice_controller.go

```diff
@@ -81,27 +82,49 @@ func (r *InstasliceReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	// 1. Ensure DaemonSet is deployed
 	daemonSet := &appsv1.DaemonSet{}
 	err := r.Get(ctx, types.NamespacedName{Name: instasliceDaemonsetName, Namespace: operatorDeployNamespace}, daemonSet)
-	if err != nil && errors.IsNotFound(err) {
-		// DaemonSet doesn't exist, so create it
-		daemonSet = createInstaSliceDaemonSet(operatorDeployNamespace)
-		err = r.Create(ctx, daemonSet)
-		if err != nil {
-			log.Error(err, "Failed to create DaemonSet")
-			return ctrl.Result{RequeueAfter: time.Minute}, err
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// DaemonSet doesn't exist, so create it
+			daemonSet = createInstaSliceDaemonSet(operatorDeployNamespace)
+			err = r.Create(ctx, daemonSet)
+			if err != nil {
+				log.Error(err, "Failed to create DaemonSet")
+				return ctrl.Result{RequeueAfter: time.Minute}, err
+			}
+			log.Info("DaemonSet created successfully, waiting for pods to be ready")
+			return ctrl.Result{RequeueAfter: requeueDelay10Sec}, nil
 		}
-		log.Info("DaemonSet created successfully, waiting for pods to be ready")
-		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
-	} else if err != nil {
 		log.Error(err, "Failed to get DaemonSet")
 		return ctrl.Result{RequeueAfter: time.Minute}, err
 	}
-	// 2. Wait for DaemonSet to be ready
-	if daemonSet.Status.DesiredNumberScheduled != daemonSet.Status.NumberReady {
-		log.Info("DaemonSet is not ready yet, waiting...")
-		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+
+	// 2. Check if at least one DaemonSet pod is ready
+	var podList v1.PodList
+	labelSelector := labels.SelectorFromSet(daemonSet.Spec.Selector.MatchLabels)
+
+	listOptions := &client.ListOptions{
+		LabelSelector: labelSelector,
+		Namespace:     operatorDeployNamespace,
+	}
+
+	if err := r.List(ctx, &podList, listOptions); err != nil {
+		log.Error(err, "Failed to list DaemonSet pods")
+		return ctrl.Result{RequeueAfter: time.Minute}, err
+	}
+	// Check if at least one daemonset pod is ready
+	for _, pod := range podList.Items {
+		if pod.Status.Phase == v1.PodRunning && len(pod.Status.ContainerStatuses) > 0 && pod.Status.ContainerStatuses[0].Ready {
+			// isAnyPodReady = true
```

Comment:
Is that supposed to be commented out? without it it appears the for loop is running for no reason.

Reply:
Yes, the purpose of the loop is to break as soon as 1 daemonset pod is ready and continue the rest of the reconciliation logic. I was using "isAnyPodReady" in later part of the code but didn't need it anymore. Let me remove that dead code,

Reply:
Thanks

Reply:
> Yes, the purpose of the loop is to break as soon as 1 daemonset pod is ready and continue the rest of the reconciliation logic.

/hold 

Shouldn't it be in while loop then?

Reply:
nevermind. 

/hold cancel

Reply:
The comment on that code says, `// Check if at least one daemonset pod is ready` 

I am not sure if just running the for loop and breaking helps here. Shouldn't it be more like this? 

```golang
// Check if at least one daemonset pod is ready
readyPodFound := false
for _, pod := range podList.Items {
    if pod.Status.Phase == v1.PodRunning && len(pod.Status.ContainerStatuses) > 0 && pod.Status.ContainerStatuses[0].Ready {
        readyPodFound = true
        break // Exit the loop once at least one ready pod is found
    }
}
```

Reply:
Or maybe you should wait here until at least one of the pod is ready before proceeding? I am just going by the comment there.

Reply:
That variable will not be used, and I was avoiding redundancy since `daemonSet.Status.NumberReady == 0` already checks. But makes sense to add that explicitly. Ok adding.